polymer/src/amp-my-embed.html

```diff
@@ -0,0 +1,285 @@
+          this.$.name.replaceChild(this.shadowRoot_, oldShadowRoot);
+        } else {
+          this.$.name.appendChild(this.shadowRoot_);
+        }
+
+        // Attach the shadow document to the new shadow root.
+        this.shadowAmp_ = amp.attachShadowDoc(this.shadowRoot_, doc, url);
+      });
+    }).catch(error => {
+      this.setState({'offline': true});
+    });
+  }
+
+  /**
+   * Cleans up internal state of current shadow AMP document.
+   * @private
+   */
+  closeShadowAmpDoc_() {
+    if (typeof this.shadowAmp_.close === 'function') {
+      this.shadowAmp_.close();
+    }
+  }
+
+  /**
+   * Hides elements (e.g. banners) that would clash with the app shell.
+   * @param {!Document} doc
+   * @private
+   */
+   hideUnwantedElementsOnDocument_(doc) {
+     const banners = doc.getElementsByClassName('banner');
+     for (let i = 0; i < banners.length; i++) {
+       banners[i].style.display = 'none';
+     }
+   }
+
+  /**
+   * Fetches and parses HTML at `url`.
+   * @private
+   * @param {string} url
+   * @return {!Promise<!Document|!string>} If fetch succeeds, resolved with {!Document}.
+   *         Otherwise, rejects with {!string} error description.
+   */
+  fetchDocument_(url) {
+    return new Promise((resolve, reject) => {
+      this.xhr_ = new XMLHttpRequest();
+      this.xhr_.open('GET', url, true);
+      this.xhr_.responseType = 'document';
+      // This is set to text/* instead of text/html because the development server
+      // only forwards requests to the proxy for requests whose 'Accept' header
+      // is NOT text/html.
+      // https://github.com/facebookincubator/create-react-app/blob/master/packages/react-scripts/template/README.md#proxying-api-requests-in-development
+      this.xhr_.setRequestHeader('Accept', 'text/*');
+      this.xhr_.onreadystatechange = () => {
+        if (this.xhr_.readyState < /* STATUS_RECEIVED */ 2) {
+          return;
+        }
+        if (this.xhr_.status < 100 || this.xhr_.status > 599) {
+          this.xhr_.onreadystatechange = null;
+          reject(new Error(`Unknown HTTP status ${this.xhr_.status}`));
+          this.xhr_ = null;
+          return;
+        }
+        if (this.xhr_.readyState === /* COMPLETE */ 4) {
+          if (this.xhr_.responseXML) {
+            resolve(this.xhr_.responseXML);
+          } else {
+            reject(new Error('No xhr.responseXML'));
+          }
+          this.xhr_ = null;
+        }
+      };
+      this.xhr_.onerror = () => { reject(new Error('Network failure')); };
+      this.xhr_.onabort = () => { reject(new Error('Request aborted')); };
+      this.xhr_.send();
+    });
+  }
+
+  /**
+   * Event listener that redirects clicks on same-domain links to react-router.
+   * This avoids page reload due to navigation from same-domain links in the AMP document,
+   * which affords seamless UX in the style of a single-page app.
+   * @private
+   * @param e {!Event}
+   */
+  clickListener_(e) {
```

Comment:
I don't see any reference to this function, is it left over from React?

Reply:
Removed, thanks